extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -769,20 +770,22 @@ protected void updateTrackMetaData() {
             String artist = null;
             String title = null;
             String album = null;
+            
             if (currentTrack != null) {
-                if (currentTrack.getAlbumArtist().equals("")) {
-                    artist = currentTrack.getCreator();
-                } else {
-                    artist = currentTrack.getAlbumArtist();
-                }
+                artist = !currentTrack.getAlbumArtist().isEmpty() ? currentTrack.getAlbumArtist()
+                        : currentTrack.getCreator();
 
                 if (!currentTrack.getTitle().contains("x-sonosapi-stream")) {
                     title = currentTrack.getTitle();
+                } else if (opmlUrl == null && currentUriMetaData != null) {
```

Comment:
what will be broken ?

The  branches are the exact same - you still have 3 options:
 1. Playing non-stream -> `currentTrack.getTitle().contains("x-sonosapi-stream")` will be `True`
 2. Playing a stream with `opmlUrl` not set -> then get the `artist` and `title` from `currentUriMetaData`
 3. Playing a stream with `opmlUrl` set -> use the `title` from the `stateMap`

In your case you just have two more brackets